### DISCUSSION TITLE
Device provisioning plugin

### DISCUSSION THREAD

**Discussion by andrewh** [netbox-community/netbox#14043] [Ideas]
Just checking on any appetite for a device provisioning plugin? Started sketching something out in code and it looks like it would link naturally to existing models like Region, Tenant, Manufacturer, Site, DeviceType, DeviceRole, Platform, etc. Would need it's own view to monitor provisioning status and history, be searchable, etc.

The plugin could add an API endpoint to emit config for a DHCP server if required (dhcpd, Kea, etc.), along with ZTP and device config based on user templates, also endpoints to handle status updates.

If there's any enthusiasm in the community I'd be happy to kick things off, maybe do some upfront design to get comments.

**Comment by chris240189** [comment]
there is an issue for that in https://github.com/miaow2/netbox-config-diff/issues

**Reply by andrewh** [reply]
Can you link the issue please?

**Reply by chris240189** [reply]
https://github.com/miaow2/netbox-config-diff/issues/25
